sklearn/preprocessing/tests/test_data.py

```diff
@@ -2377,3 +2378,190 @@ def test_power_transformer_lambda_zero():
     pt.lambdas_ = np.array([0])
     X_trans = pt.transform(X)
     assert_array_almost_equal(pt.inverse_transform(X_trans), X)
+
+
+def test_count_featurizer():
+    # test count featurizer fit-transform on a very standard data
+
+    X = np.array([[0, 2, 1], [1, 0, 3], [1, 0, 2], [1, 0, 2]])
+    cf_1 = CountFeaturizer()
+    assert_array_equal(
+        cf_1.fit_transform(X),
+        np.array([[0, 2, 1, 1], [1, 0, 3, 1], [1, 0, 2, 2], [1, 0, 2, 2]]))
+
+
+def test_count_featurizer_diff_fit():
+    # test count featurizer where fit is trained on something different
+    # than what we are transforming
+
+    X = np.array([[0, 2, 1], [1, 0, 3], [1, 0, 2], [1, 0, 2]])
+    X2 = np.array([[0, 2, 1], [0, 2, 1], [0, 2, 1], [1, 0, 2]])
+    cf_1 = CountFeaturizer().fit(X)
+    assert_array_equal(
+        cf_1.transform(X2),
+        np.array([[0, 2, 1, 1], [0, 2, 1, 1], [0, 2, 1, 1], [1, 0, 2, 2]]))
+
+
+def test_count_featurizer_diff_fit_y():
+    # test count featurizer where fit is trained on something different
+    # than what we are transforming with a y in fit
+
+    X = np.array([[0, 2, 1], [1, 0, 3], [1, 0, 2], [1, 0, 2]])
+    X2 = np.array([[0, 2, 1], [0, 2, 1], [0, 2, 1], [1, 0, 2]])
+    y = np.array([0, 0, 1, 0])
+    cf_1 = CountFeaturizer().fit(X, y=y)
+    assert_array_equal(
+        cf_1.transform(X2),
+        np.array([[0, 2, 1, 1, 0], [0, 2, 1, 1, 0],
+                 [0, 2, 1, 1, 0], [1, 0, 2, 1, 1]]))
+
+
+def test_count_featurizer_diff_fit_multi_y():
```

Comment:
* test_count_featurizer_diff_fit_multi_y: it is unclear to me why the output has 3 additional values.
I see 1 x value, 2 y values, so 2^2 = 4 potential 4 combinations (and 2 actually occurring combinations).
Please explain in comments